### PR TITLE
1093 bug wrong docker files for library microservice

### LINKED ISSUE #1093: [BUG] Wrong docker files for library microservice
### Describe the bug

The docker files of libms ([from npm](https://github.com/INTO-CPS-Association/DTaaS/blob/feature/distributed-demo/docker/libms.npm.dockerfile) and [direct build](https://github.com/INTO-CPS-Association/DTaaS/blob/feature/distributed-demo/docker/libms.dockerfile)) are using incorrect configuration.

### To Reproduce

1.  Pull the libms image `docker pull intocps/libms`
1. Use instructions in [DOCKER](https://github.com/INTO-CPS-Association/DTaaS/blob/feature/distributed-demo/servers/lib/DOCKER.md) to run libms container
1. Go to `http://localhost:4001/lib`
1. See error

### Expected behavior

The microservice must serve files using the latest yaml configuration. It should also support git mode.

### PR DESCRIPTION
**Pull request by nichlaes**
## Type of Change

- [X] Bug fix
- [X] Documentation update

## Description

Fixes

#1093
#1095 

## Testing

Will be tested through docker github action.

## Impact

Github actions should work, and readme is updated to use new yaml config

## Checklist

- [X] My code adheres to the coding and style guidelines of the project.
- [X] I have added tests for all the new code and any changes made to
      existing code.
- [X] I have made corresponding changes to the documentation.

### CODE REVIEW

**Comment by prasadtalasila** [comment]
@nichlaes please rebase to `feature/distributed-demo` branch

File: docker/compose.dev.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ services:
     restart: unless-stopped
     volumes:
       - ${DTAAS_DIR}/files:/dtaas/libms/files
-      - ${LIB_CONFIG}:/dtaas/libms/config/.env.default
+      - ${LIB_CONFIG}:/dtaas/libms/config/libms.dev.yaml
```
Comment (prasadtalasila) [inline docker/compose.dev.yml]: wouldn't it be 

```
- ${LIB_CONFIG}:/dtaas/libms/config/libms.yaml
```

File: servers/lib/DEVELOPER.md [hunk 2]
```diff
@@ -140,7 +130,7 @@ docker compose -f compose.lib.dev.yml up -d
 
 This command brings up the lib docker container and makes
 the website available at <http://localhost:4001>.
-The `config/.env.default` file is used as the microservice configuration.
+The `config/libms.dev.yaml` file is used as the microservice configuration.
```
Comment (prasadtalasila) [inline servers/lib/DEVELOPER.md:139]: The volume mapping of `servers/lib/compose.lib.dev.yml` need to be updated to reflect this use.

File: servers/lib/DEVELOPER.md
Comment (prasadtalasila) [inline servers/lib/DEVELOPER.md:1]: please add the use of git mode to DOCKER.md

File: servers/lib/config/libms.yaml.default
Comment (prasadtalasila) [inline servers/lib/config/libms.yaml.default:1]:
```
port: '4001'
mode: 'local'    #git or local
local-path: '..\..\files'
log-level: 'debug'
apollo-path: '/lib'
graphql-playground: 'true'

git-repos:   #only used in git mode
  - user-1: 
      repo-url: 'https://github.com/INTO-CPS-Association/DTaaS.git'
```

Comment (prasadtalasila) [reply servers/lib/config/libms.yaml.default:1]: I checked it before, but please do run libms once with this new config file containing comments.

File: servers/lib/config/libms.yaml.default [hunk 3]
```diff
@@ -1,12 +1,10 @@
...
 apollo-path: '/lib'
 graphql-playground: 'true'
 
 git-repos:
   - user-1: 
-      repo-url: 'https://github.com/isomorphic-git/lightning-fs'
-  - user-2: 
-      repo-url: 'https://github.com/isomorphic-git/lightning-fs'
\ No newline at end of file
+      repo-url: 'https://github.com/INTO-CPS-Association/DTaaS.git'
```
Comment (prasadtalasila) [inline servers/lib/config/libms.yaml.default]: `repo-url: 'https://gitlab.com/dtaas/user1.git'`

File: docker/libms.npm.dockerfile
Comment (prasadtalasila) [inline docker/libms.npm.dockerfile:1]: please update the base image to `node:22.12.0-slim`

**Review by prasadtalasila — COMMENTED** [review]
@nichlaes please see the comments.

File: servers/lib/config/libms.yaml.default (hunk 3 above)
Comment (prasadtalasila) [reply servers/lib/config/libms.yaml.default]: @nichlaes please use the following yaml for `libms.yml.default`

```yaml
port: '4001'
mode: 'local'    #git or local
local-path: '..\..\files'
log-level: 'debug'
apollo-path: '/lib'
graphql-playground: 'true'

git-repos:   #only used in git mode
  - user1: 
      repo-url: 'https://gitlab.com/dtaas/user1.git'
  - user2: 
      repo-url: 'https://gitlab.com/dtaas/user2.git'
  - common: 
      repo-url: 'https://gitlab.com/dtaas/common.git'
```

File: servers/lib/DEVELOPER.md (hunk 2 above)
Comment (nichlaes) [reply servers/lib/DEVELOPER.md:139]: @prasadtalasila  But I guess this depends on the ${LIB_CONFIG}?

File: servers/lib/DEVELOPER.md
Comment (nichlaes) [reply servers/lib/DEVELOPER.md:1]: Would that not be duplicate of documentation? There's no documentation in the file for the other configurations, port, path etc. or the mode.

File: servers/lib/DEVELOPER.md (hunk 2 above)
Comment (prasadtalasila) [reply servers/lib/DEVELOPER.md:139]: @nichlaes we do not have any environment variables for the `compose.lib.dev.yml`. So it has to be hard coded.

File: servers/lib/DEVELOPER.md
Comment (prasadtalasila) [reply servers/lib/DEVELOPER.md:1]: I agree that it is a duplication of information. Would you rather provide a link to `servers/lib/README.md#gear-configure`?

File: servers/lib/config/libms.yaml.default (hunk 3 above)
Comment (prasadtalasila) [reply servers/lib/config/libms.yaml.default]: It makes sense to use the same in README.md as well. We are using convention of **user1, user2, common** etc. So being consistent in configuration makes sense as well.

File: servers/lib/DEVELOPER.md (hunk 2 above)
Comment (nichlaes) [reply servers/lib/DEVELOPER.md:139]: Should I hardcode it in the docker compose file?

Comment (prasadtalasila) [reply servers/lib/DEVELOPER.md:139]: Please do

File: docker/compose.dev.yml
Comment (prasadtalasila) [inline docker/compose.dev.yml:1]: @nichlaes 
In `docker/.env` file, can we remove the following lines?

```INI
OAUTH_CONF_FILEPATH='/Users/<Username>/DTaaS/docker/conf.dev'
LIB_CONFIG='/Users/<Username>/DTaaS/servers/lib/config/.env.default'
```

File: servers/lib/config/libms.yaml.default (hunk 3 above)
Comment (prasadtalasila) [reply servers/lib/config/libms.yaml.default]: please add the same structure to `config/libms.dev.yml`

**Pull request merged**